hosting.md

```diff
@@ -0,0 +1,40 @@
+---
+title: Hosting
+layout: recommendation
+expires: 2017-11-01
+---
+
+GDS follows the [Government Cloud First policy](https://www.gov.uk/guidance/government-cloud-first-policy) meaning that we prefer the use of PaaS and IaaS solutions over managing our own hardware
+
+## User Needs
+
+* Users want systems that are highly scalable and available
+* GDS wants to manage the environments easily and simply, with confidence via automated tools
+
+## Principles
+
+GDS will build new tools on a primary cloud platform for each of PaaS and IaaS.
+
+We prefer systems built on a PaaS to systems built on IaaS where possible.
+
+We prefer cloud suppliers that are well supported by common tooling to reduce cost of migrating to a new supplier.
+
+## Tools
+
+GDS has selected two cloud platforms:
+
+### PaaS
+
+* GOV.UK PaaS
+  https://cloud.service.gov.uk 
+
+### IaaS
+
+* Amazon Web Services
+  https://aws.amazon.com
+
+### Lock in concerns
+
+In order to minimise lock-in, we determine that for the IaaS model, we will use only the core features of AWS that are common with the biggest competitors.  That is to say we will be using  for example: Compute, Networking (load balancing and network segregation), File storage (EBS and S3), Hosted databases (RDS and DynamoDB).
```

Comment:
can we phrase things in terms of switching costs instead of lock in?

I think "lock in" is a poor term. Firstly, it leads people to think in binary terms: "we are/aren't locked in", instead of "it'll cost us a little/a lot to switch provider". Secondly, it leads people to divide services into two buckets: lock-in-prone services and generic services, and that the way to avoid lock-in is to avoid those lock-in-prone services.

But what happens then?  For example: a team is told that they can't use hosted elasticsearch, because it's not a common enough service, so they build, host and manage their own elasticsearch.  What they've done is increased the total size of their self-managed infrastructure, which increases the switching costs of moving that infrastructure to another provider.  The whole exercise was self-defeating.

Switching costs need to be viewed as a property of the whole system.  Trying to take a shortcut by making a list of "approved" vs "unapproved" services leads to poor decision making, sprawling and poorly-factored infrastructure, and ultimately greater switching costs and greater lock in.